### ISSUE TITLE
Floating navigation

### ISSUE THREAD

**Issue by kyleshay** [DestinyItemManager/DIM#540]
> I notice I'm always having to scroll a lot since I have so many items in all of the categories. I'd love to see a side bar fixed to the right of the screen with a button for each item category that would jump you to the section that you click.

[Source](https://www.reddit.com/r/DestinyItemManager/comments/4k8tge/i_notice_im_always_having_to_scroll_a_lot_since_i/)

Something like this: http://www.onextrapixel.com/examples/smooth-jump-to-sub-navigation-menu-in-one-js-call/

Maybe instead of it always being visible, it is like a drawer that collapses into the edge of the screen. Hovering the menu, it expands and you can click any section... mouse out it re-collapses.

Maybe it would be unnecessary if we consider #401

**Comment by bhollis** [comment]
I was just thinking about this the other day. I have some ideas.

Could we get some cool icons to represent weapons, armor, progress, etc?

**Comment by bhollis** [comment]
If somebody volunteers some awesome icons, I'll do this. My thought is to have a fixed menu along the left side, or in the header.

**Comment by ericnelson0** [comment]
I've been messing around with this in a local branch. What would you guys think about something like this?

![example_nav](https://cloud.githubusercontent.com/assets/4594931/15682777/e556c1d8-2724-11e6-9c2a-1566a8120c6d.gif)

The icons are pretty bad IMO, but that's not something I'm good at.

**Comment by kyleshay** [comment]
that's super awesome! we could even save the open/close state of it between sessions. that way if people prefer it open they don't need to toggle it on every load. 

wondering what would be the case if the window height is smaller than the number of items in that list, it would then be scroll-able, two columns side-by-side? 

should we indicate which section they are scrolled to by highlighting the icon?

wondering if we should go with another logo versus the hamburger icon? unless we wanted to migrate some additional settings there?

just throwing out some initial thoughts now that we have something to go off of.

**Comment by ericnelson0** [comment]
@kyleshay I made the heights of the icons dependent on the height of the browser, so it should just make the icons smaller.

Would we update the highlighted section depending on where they are currently scrolled or remove all highlighting once they have scrolled at all?

Not sure about the icon, maybe a triangle pointing right?

**Comment by kyleshay** [comment]
triangle pointing a direction works. if it doesn't translate well i'm fine with sticking with the hamburger. 

yeah ideally if the user scrolls the page, the row of items at the top of the page would select that icon. does that make sense?

**Comment by ericnelson0** [comment]
Yeah that makes sense. It can be a little tricky if the window width is squeezed because the rows can be at different heights on the page, but it shouldn't be too much of an issue.

Does anyone have any ideas about the icons? I would still  like to see some better ones.

**Comment by kyleshay** [comment]
... heres a thought, should we just always keep it expanded? it doesn't take up that much space and it provides a decent amount of value. could go either way. 

i'll put a message out about the icons.

**Comment by MyrHerder** [comment]
Behavior looks good. I have no problem with the icons. Right arrow and remembered state would be nice.

**Comment by kyleshay** [comment]
just an aside, here is an alternative approach that i stood up a year ago, the original thought was it would really help on mobile devices and would very similar to how on iOS devices they would have `a -> z` for musical artists (as an example) and you could quickly jump to `H`, or whatever.

src: https://github.com/kyleshay/scrollnav.js
demo: https://kyleshay.github.io/scrollnav.js/
use in a responsive app: http://kyleshay.github.io/demo/

I do like the approach we have here with the icons and just click to scroll, but figured link this just a reference.

**Comment by bhollis** [comment]
I think that's pretty cool! I was thinking of doing icons just for the "major" sections rather than each individual one. We might also want to have the icons be smaller or a bit simpler.

Also, Bootstrap has a script (and I'm sure there are more like it) that can select items in a menu based on where you scroll.

**Comment by kyleshay** [comment]
https://www.reddit.com/r/DestinyItemManager/comments/4rsgp1/idea_on_the_bar_with_the_characters_put_shortcuts/

**Comment by kyleshay** [comment]
I think if we end up with something that is similar to what eric has in his mockup, with the section at the top of the page highlighted on the left this would be neat.

another thing to consider is if a section is collapsed, it should be hidden from the quickscroll column. 

Suppose the only thing to decide is if we want to do it by major section or by sub-section. for mobile sub-section could be nice (but it could also be a challenge to show all of the icons in a column like that on mobile, unless it also scrolls.)
